samples/bootloader/src/main.c

```diff
@@ -109,8 +109,9 @@ static void boot_from(const struct fw_info *fw_info)
 		__set_CONTROL(__get_CONTROL() & ~CONTROL_SPSEL_Msk);
 	}
 
-	__DSB(); /* Force Memory Write before continuing */
-	__ISB(); /* Flush and refill pipeline with updated permissions */
+	/* Flush and refill pipeline with updated permissions and SP */
+	__DSB();
+	__ISB();
 
 	VTOR = fw_info->address;
```

Comment:
ISB, DSB is actully needed, here, before writing to SCB_VTOR, to make sure the new VTOR is recognized by all subsequent instructions

Reply:
Theoretically, fw_info.address might be cached in stack, and you might have changed stack pointer on line 109, so this seems (at first sight) a hidden danger

Reply:
Is this addressed by the change where the vector table is set after we check the SP